src/containerapp/azext_containerapp/custom.py

```diff
@@ -1244,16 +1317,27 @@ def containerapp_up(cmd,
                     user_assigned=None,
                     system_assigned=None,
                     custom_location_id=None,
-                    connected_cluster_id=None):
+                    connected_cluster_id=None,
+                    model_registry=None,
+                    model_name=None,
+                    model_version=None):
     from ._up_utils import (_validate_up_args, _validate_custom_location_connected_cluster_args, _reformat_image, _get_dockerfile_content, _get_ingress_and_target_port,
                             ResourceGroup, Extension, CustomLocation, ContainerAppEnvironment, ContainerApp, _get_registry_from_app,
                             _get_registry_details, _get_registry_details_without_get_creds, _create_github_action, _set_up_defaults, up_output,
-                            check_env_name_on_rg, get_token, _has_dockerfile)
+                            check_env_name_on_rg, get_token, _has_dockerfile, _validate_azml_args, _is_azml_app, _validate_azml_env_and_create_if_needed, _set_azml_env_vars)
     from azure.cli.command_modules.containerapp._github_oauth import cache_github_token
     HELLOWORLD = "mcr.microsoft.com/k8se/quickstart"
+    ACA_AZML_MCR = "mcr.microsoft.com/k8se/aca-foundry-model-host"
     dockerfile = "Dockerfile"  # for now the dockerfile name must be "Dockerfile" (until GH actions API is updated)
 
     register_provider_if_needed(cmd, CONTAINER_APPS_RP)
+    is_azureml_app = _is_azml_app(model_registry, model_name, model_version)
+    model_asset_id = None
+    model_reference_endpoint = None
+    if is_azureml_app:
+        model_asset_id, model_reference_endpoint, model_type, model_load_class, image = _validate_azml_args(cmd, ACA_AZML_MCR, image, model_registry, model_name, model_version)
+        env_vars = _set_azml_env_vars(cmd, env_vars, model_asset_id, model_reference_endpoint, model_type, model_load_class, is_azml_mcr_app=True if image.lower() == ACA_AZML_MCR.lower() else False)
```

Comment:
Style error:
src/containerapp/azext_containerapp/custom.py:1339:141: R1719: The if expression can be replaced with 'test' (simplifiable-if-expression)

```suggestion
        env_vars = _set_azml_env_vars(cmd, env_vars, model_asset_id, model_reference_endpoint, model_type, model_load_class, is_azml_mcr_app=image.lower() == ACA_AZML_MCR.lower())
```